Failed to load Huggingface native library:  Unexpected flavor: cpu

I fail to execute the use case "Search the Quarkus docs for how to create a REST endpoint"

the issue I see in the logs of the agent are:
```
2026-04-24 08:22:30,186 INFO  [io.quarkus.agent.mcp.ContainerManager] (executor-thread-1) Starting pgvector container with Quarkus latest docs (ghcr.io/quarkusio/chappie-ingestion-quarkus:latest)...                                                                                                        
2026-04-24 08:22:30,253 INFO  [tc.ghcr.io/quarkusio/chappie-ingestion-quarkus:latest] (executor-thread-1) Creating container for image: ghcr.io/quarkusio/chappie-ingestion-quarkus:latest                                                                                                                    
2026-04-24 08:22:30,449 INFO  [org.testcontainers.utility.RegistryAuthLocator] (executor-thread-1) Credential helper/store (docker-credential-desktop) does not have credentials for ghcr.io                                                                                                                  
2026-04-24 08:22:30,587 INFO  [tc.ghcr.io/quarkusio/chappie-ingestion-quarkus:latest] (executor-thread-1) Reusing container with ID: ce42ba33352c2cf629d77c299bbcc29b1663579fe99b46b400bde032a1f8da45 and hash: 91806ec455efb498c98253cea5dc02a5261551c4                                                      
2026-04-24 08:22:30,588 INFO  [tc.ghcr.io/quarkusio/chappie-ingestion-quarkus:latest] (executor-thread-1) Reusing existing container (ce42ba33352c2cf629d77c299bbcc29b1663579fe99b46b400bde032a1f8da45) and not creating a new one                                                                            
2026-04-24 08:22:30,810 INFO  [tc.ghcr.io/quarkusio/chappie-ingestion-quarkus:latest] (executor-thread-1) Container ghcr.io/quarkusio/chappie-ingestion-quarkus:latest started in PT0.557183S                                                                                                                 
2026-04-24 08:22:30,811 INFO  [io.quarkus.agent.mcp.ContainerManager] (executor-thread-1) pgvector container started for Quarkus latest (mapped port: 38849)                                                                                                                                                  
2026-04-24 08:22:30,836 INFO  [io.quarkus.agent.mcp.DocSearchTools] (executor-thread-1) Connecting to pgvector at localhost:38849 (Quarkus default)... 
2026-04-24 08:22:30,852 INFO  [io.quarkus.agent.mcp.DocSearchTools] (executor-thread-1) Connected to pgvector embedding store.                         
2026-04-24 08:22:30,854 INFO  [io.quarkus.agent.mcp.EmbeddingModelProducer] (executor-thread-1) Loading BGE Small EN v1.5 embedding model (384 dimensions)...                                                                                                                                                 
2026-04-24 08:22:35,467 INFO  [ai.djl.util.Platform] (executor-thread-1) Found matching platform from: jar:file:/Users/vsevel/dev/github/quarkus-agent-mcp/target/quarkus-app/lib/main/ai.djl.huggingface.tokenizers-0.31.1.jar!/native/lib/tokenizers.properties                                             
2026-04-24 08:22:35,469 INFO  [ai.djl.huggingface.tokenizers.jni.LibUtils] (executor-thread-1) library not found in classpath: native/lib/osx-x86_64/cpu/libtokenizers.dylib     
2026-04-24 08:22:35,476 ERROR [io.quarkiverse.mcp.server.runtime.ToolMessageHandler] (executor-thread-1) Unable to call tool quarkus_searchDocs: java.lang.ExceptionInInitializerError                                                                                                                        
        at io.quarkus.agent.mcp.EmbeddingModelProducer.embeddingModel(EmbeddingModelProducer.java:22)                                                  
        at io.quarkus.agent.mcp.EmbeddingModelProducer_ProducerMethod_embeddingModel_fX4zDORhnjWo9ysyBbYuCUhd50c_Bean.create(Unknown Source)           
        at io.quarkus.agent.mcp.EmbeddingModelProducer_ProducerMethod_embeddingModel_fX4zDORhnjWo9ysyBbYuCUhd50c_Bean.create(Unknown Source)           
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)                                              
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)                                                              
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)                                                              
        at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.ch4(Unknown Source)                          
        at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)              
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)                                                                
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:23)                                                       
        at dev.langchain4j.model.embedding.EmbeddingModelProducer_ProducerMethod_embeddingModel_fX4zDORhnjWo9ysyBbYuCUhd50c_ClientProxy.arc$delegate(Unknown Source)                                                                                                                                          
        at dev.langchain4j.model.embedding.EmbeddingModelProducer_ProducerMethod_embeddingModel_fX4zDORhnjWo9ysyBbYuCUhd50c_ClientProxy.embed(Unknown Source)                                                                                                                                                 
        at io.quarkus.agent.mcp.DocSearchTools.searchDocs(DocSearchTools.java:121)                                                                     
        at io.quarkus.agent.mcp.DocSearchTools_searchDocs_Invoker_HfHnkqzbR2g-MUUWIu54La1Rmbk.invoke(Unknown Source)                                   
        at io.quarkus.agent.mcp.DocSearchTools_searchDocs_LazyInvoker_HfHnkqzbR2g-MUUWIu54La1Rmbk.invoke(Unknown Source)                               
        at io.quarkiverse.mcp.server.runtime.FeatureManagerBase$FeatureMetadataInvoker.call(FeatureManagerBase.java:442)                               
        at io.quarkiverse.mcp.server.runtime.FeatureManagerBase$1.lambda$call$0(FeatureManagerBase.java:108)                                           
        at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)                                              
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:68)                                                                                                                                                
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)           
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:42)              
        at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)                                  
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)                                                                     
        at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)                                         
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)                                                                     
        at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap.subscribe(UniOnItemOrFailureFlatMap.java:27)                                     
        at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)                                                                     
        at io.smallrye.mutiny.groups.UniSubscribe.withSubscriber(UniSubscribe.java:51)                                                                 
        at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:110)                                                                          
        at io.smallrye.mutiny.groups.UniSubscribe.with(UniSubscribe.java:88)                                                                           
        at io.quarkiverse.mcp.server.runtime.FeatureManagerBase$3.call(FeatureManagerBase.java:356)                                                    
        at io.quarkiverse.mcp.server.runtime.FeatureManagerBase$3.call(FeatureManagerBase.java:352)                                                    
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$4(ContextImpl.java:192)                                                               
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:270)                                                                       
        at io.vertx.core.impl.ContextImpl$1.execute(ContextImpl.java:221)                                                                              
        at io.vertx.core.impl.WorkerTask.run(WorkerTask.java:56)                                                                                       
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:677)                                                      
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)                                                     
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)                                                           
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)                                                      
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)                                                     
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)                                                                        
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)                                                    
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)                                                       
        at java.base/java.lang.Thread.run(Thread.java:1583)                                                                                            
Caused by: java.lang.RuntimeException: ai.djl.engine.EngineException: Failed to load Huggingface native library.                                       
        at dev.langchain4j.model.embedding.onnx.OnnxBertBiEncoder.<init>(OnnxBertBiEncoder.java:53)                                                    
        at dev.langchain4j.model.embedding.onnx.AbstractInProcessEmbeddingModel.loadFromJar(AbstractInProcessEmbeddingModel.java:42)                   
        at dev.langchain4j.model.embedding.onnx.bgesmallenv15q.BgeSmallEnV15QuantizedEmbeddingModel.<clinit>(BgeSmallEnV15QuantizedEmbeddingModel.java:30)                                                                                                                                                    
        ... 45 more                                                                                                                                    
Caused by: ai.djl.engine.EngineException: Failed to load Huggingface native library.                                                                   
        at ai.djl.huggingface.tokenizers.jni.LibUtils.<clinit>(LibUtils.java:56)                                                                       
        at ai.djl.huggingface.tokenizers.HuggingFaceTokenizer.newInstance(HuggingFaceTokenizer.java:186)                                               
        at dev.langchain4j.model.embedding.onnx.OnnxBertBiEncoder.<init>(OnnxBertBiEncoder.java:50)                                                    
        ... 47 more                                                                                                                                    
Caused by: ai.djl.engine.EngineException: Unexpected flavor: cpu                                                                                       
        at ai.djl.huggingface.tokenizers.jni.LibUtils.copyJniLibrary(LibUtils.java:200)                                                                
        at ai.djl.huggingface.tokenizers.jni.LibUtils.loadLibrary(LibUtils.java:87)                                                                    
        at ai.djl.huggingface.tokenizers.jni.LibUtils.<clinit>(LibUtils.java:54)                                                                       
        ... 49 more                      
```

I am running on macos.
to make it easy to see the agent's behavior I have added dependency: `quarkus-mcp-server-http`, and registered the agent with claude using: `claude mcp add --transport http quarkus-agent http://localhost:18080/mcp`, and started it with: `java -Dquarkus.http.port=18080 -jar target/quarkus-app/quarkus-run.jar`

seems to be related to https://github.com/langchain4j/langchain4j/issues/2144
I downgraded the tokenizer to `0.28`, and it worked

--- a/pom.xml
+++ b/pom.xml
@@ -54,6 +54,7 @@
         <langchain4j.version>1.12.2-beta22</langchain4j.version>
         <skipITs>true</skipITs>
         <surefire-plugin.version>3.5.4</surefire-plugin.version>
+        <djl.version>0.28.0</djl.version>
         <testcontainers.version>2.0.5</testcontainers.version>
     </properties>
 
@@ -65,6 +66,18 @@
                 <version>${quarkus.platform.version}</version>
                 <type>pom</type>
                 <scope>import</scope>
+            </dependency>
+            <!-- Pin DJL to 0.28.x: version 0.31.1 (pulled by langchain4j) fails to load
+                 the Huggingface tokenizer native library on macOS with "Unexpected flavor: cpu" -->
+            <dependency>
+                <groupId>ai.djl</groupId>
+                <artifactId>api</artifactId>
+                <version>${djl.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>ai.djl.huggingface</groupId>
+                <artifactId>tokenizers</artifactId>
+                <version>${djl.version}</version>
             </dependency>
         </dependencies>
     </dependencyManagement>

